src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -262,18 +276,19 @@ class DeepDiveLogParser extends JavaTokenParsers {
     }
 
   def constantWeight = floatingPointNumberAsDouble ^^ {   KnownFactorWeight(_) }
-  def unknownWeight  = repsep(variableName, ",")   ^^ { UnknownFactorWeight(_) }
+  def unknownWeight  = repsep(expr, ",")   ^^ { UnknownFactorWeight(_) }
```

Comment:
If we support expr for `unknownWeight` here, why do we need `constantWeight`?  Isn't it more natural for the compiler to recognize constant expr (with int/float) and decide there?